### PR TITLE
add recalc_center

### PR DESCRIPTION
**Pull request by dorlib**
add the function for recalculating the center of a cluster 

also, filename should be "kmeans.py"

please go over the third commit, iv'e fixed bugs there and changed *coord to coord because it was problematic and I'm not sure if it necessary, need to think about it 
in the meanwhile, i've managed to run the program and and the results was as expected, but with a little offset in the third char after the dot, need to check it out

### CODE REVIEW

File: kmeans.py [hunk 1]
```diff
@@ -28,12 +28,25 @@ def add(self, p):
         self.members.append(p)
         return self.recalc_center()
     
-    def recalc_center()->float:
+    def recalc_center(self)->float:
         '''returns eclidean Distance, between the updated centroid to the previous one'''
-        pass
+        coords = () * self.center.dimention
```
Comment (AfikBenShimol) [inline kmeans.py:33]: this returns an empty tuple. If you want an empty tuple the size of the dimension, you can use 
`coords = tuple((None for i in range(self.center.dimention)))`

File: kmeans.py [hunk 2]
```diff
@@ -28,12 +28,25 @@ def add(self, p):
...
+        for i in range(self.center.dimention):
+            sum = 0
+            for point in self.members:
+                sum += point.coord[i]
+            
+            coords = coords + (sum / len(self.members), )
```
Comment (AfikBenShimol) [inline kmeans.py:39]: This leaves an empty spot after the last coordinate is calculated. 
Meaning we'll get:
`(x0, x1,..., xn,   )`
`_________________↑`
This does not show up when printing the coordinates but it takes memory for no reason

**Review by AfikBenShimol — CHANGES_REQUESTED** [review]
Great work, a few fixes needed and I think we can optimize it a bit if we want, but it's almost finished.

File: kmeans.py (hunk 1 above)
Comment (dorlib) [reply kmeans.py:33]: when I do the change you've suggested it doesn't work, you want to try it ?

File: kmeans.py (hunk 2 above)
Comment (dorlib) [reply kmeans.py:39]: when I do the change you've suggested it doesn't work, you want to try it ?

**Pull request merged**